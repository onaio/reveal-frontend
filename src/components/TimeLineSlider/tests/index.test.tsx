import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { TimelineSlider, TimelineSliderProps } from '..';

describe('src/components/TimeLineSlider', () => {
  it('works correctly', () => {
    const props: TimelineSliderProps = {
      keyOfCurrentStop: '0',
      stops: [
        {
          keys: '1',
          labelBelowStop: 'AUTO_TARGET_JURISDICTIONS_BY_RISK',
          labelInStop: '1',
        },
        {
          keys: ['2', '3'],
          labelBelowStop: 'REFINE_SELECTED_JURISDICTIONS',
          labelInStop: '2',
        },
      ],
    };
    const wrapper = mount(<TimelineSlider {...props} />);
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"AUTO_TARGET_JURISDICTIONS_BY_RISKREFINE_SELECTED_JURISDICTIONS"`
    );
    wrapper.find('li').forEach(li => {
      expect(toJson(li)).toMatchSnapshot();
    });
  });
});
