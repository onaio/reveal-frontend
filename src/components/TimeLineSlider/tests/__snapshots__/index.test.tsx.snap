// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/TimeLineSlider works correctly 2`] = `
<li
  className=""
  data-step="1"
  key="0-step-slider-stop"
>
  AUTO_TARGET_JURISDICTIONS_BY_RISK
</li>
`;

exports[`src/components/TimeLineSlider works correctly 3`] = `
<li
  className=""
  data-step="2"
  key="1-step-slider-stop"
>
  REFINE_SELECTED_JURISDICTIONS
</li>
`;
